components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -294,17 +299,24 @@ void CosmeticFiltersJSHandler::OnShouldDoCosmeticFiltering(
   cosmetic_filters_resources_->UrlCosmeticResources(
       url_.spec(),
       base::BindOnce(&CosmeticFiltersJSHandler::OnUrlCosmeticResources,
-                     base::Unretained(this)));
+                     base::Unretained(this), std::move(callback)));
+}
+
+void CosmeticFiltersJSHandler::OnUrlCosmeticResources(
+    base::OnceClosure callback,
+    base::Value result) {
+  resources_dict_ = base::DictionaryValue::From(
+      base::Value::ToUniquePtrValue(result.Clone()));
```

Comment:
Clone could be expensive, wouldn't `std::move(result)` work?

Reply:
that's a good point and I think so